### PR TITLE
feat: add env var file support

### LINKED ISSUE #32: Add env_vars_file option
### TL;DR

It does not seem possible to generate environment values dynamically to pass into the `env_vars` input.

A great improvement would be to make this possible by either supporting the use of a YAML config file like the `gcloud` CLI does, and/or supporting using GitHub Actions' `set-output` to pass a dynamic string in `KEY=VALUE,KEY2=VALUE2` format.

### Design

**Action YAML**

New `env_vars_file` input.

### PR DESCRIPTION
**Pull request by bharathkkb**
Adds support for an env var yaml file input similar to [gcloud cli ](https://cloud.google.com/sdk/gcloud/reference/functions/deploy#--env-vars-file)
fixes #32 #17

### CODE REVIEW

File: src/cloudFunction.ts [hunk 1]
```diff
@@ -108,12 +111,24 @@ export class CloudFunction {
...
     let envVars;
     if (opts?.envVars) {
       envVars = this.parseEnvVars(opts.envVars);
       request.environmentVariables = envVars;
     }
+    if (opts?.envVarsFile) {
+      envVars = this.parseEnvVarsFile(opts.envVarsFile);
+      request.environmentVariables = envVars;
+    }
+    console.log(envVars);
```
Comment (averikitsch) [inline src/cloudFunction.ts:131]: nit: remove console.log

File: .github/workflows/deploy-cf-it.yml [hunk 2]
```diff
@@ -140,6 +140,7 @@ jobs:
         credentials: ${{ secrets.DEPLOY_CF_SA_KEY_JSON }}
         event_trigger_type: providers/cloud.pubsub/eventTypes/topic.publish
         event_trigger_resource: ${{ secrets.DEPLOY_CF_EVENT_PUBSUB_TOPIC }}
+        env_vars_file: './tests/env-var-files/test.good.yaml'
```
Comment (averikitsch) [inline .github/workflows/deploy-cf-it.yml:143]: No Integration tests? It looks like the other jobs run e2e tests

File: README.md [hunk 3]
```diff
@@ -64,6 +64,8 @@ steps:
   KEY1=VALUE1,KEY2=VALUE2. All existing environment variables will be
   removed first.
 
+- `env_vars_file`: (Optional) Path to a local YAML file with definitions for all environment variables. Only one of env_vars or env_vars_file can be specified.
```
Comment (averikitsch) [inline README.md:67]: Might be nice to link to the example file `tests/env-var-files/test.good.yaml`

**Review by averikitsch — APPROVED** [review]

File: .github/workflows/deploy-cf-it.yml (hunk 2 above)
Comment (bharathkkb) [reply .github/workflows/deploy-cf-it.yml:143]: These are actually the Integration tests (we build dist and use the built action)  however I need to split the [unit](https://github.com/google-github-actions/deploy-cloud-functions/blob/main/tests/cloudFunction.test.ts) from [e2e](https://github.com/google-github-actions/deploy-cloud-functions/blob/main/tests/cloudFunctionClient.test.ts).

File: src/cloudFunction.ts (hunk 1 above)
Comment (bharathkkb) [reply src/cloudFunction.ts]: oops done

File: README.md (hunk 3 above)
Comment (bharathkkb) [reply README.md]: good idea, done and added a json example

**Pull request merged**